src/snowflake/snowpark/udf.py

```diff
@@ -602,7 +628,7 @@ def register(
         - :meth:`register_from_file`
         """
         with open_telemetry_udf_context_manager(self.register, func=func, name=name):
-            if not callable(func):
+            if not callable(func) and kwargs.get("_registered_object_name") is None:
                 raise TypeError(
```

Comment:
is this AST only change?

Reply:
@sfc-gh-evandenberg has some more context on this. Basically, this a non-public parameter which helps to replay ASTs.